Modules/Core/Common/src/itkPoolMultiThreader.cxx

```diff
@@ -148,13 +138,14 @@ PoolMultiThreader::SingleMethodExecute()
   // Now, the parent thread calls this->SingleMethod() itself
   m_ThreadInfoArray[0].UserData = m_SingleData;
   m_ThreadInfoArray[0].NumberOfWorkUnits = m_NumberOfWorkUnits;
-  ExceptionHandler exceptionHandler([this] { m_SingleMethod(&m_ThreadInfoArray[0]); });
+  ExceptionHandler exceptionHandler;
+  exceptionHandler.TryAndCatch([this] { m_SingleMethod(&m_ThreadInfoArray[0]); });
 
   // The parent thread has finished SingleMethod()
   // so now it waits for each of the other work units to finish
   for (threadLoop = 1; threadLoop < m_NumberOfWorkUnits; ++threadLoop)
   {
-    exceptionHandler.TryToGetFuture(m_ThreadInfoArray[threadLoop]);
+    exceptionHandler.TryAndCatch([&] { m_ThreadInfoArray[threadLoop].Future.get(); });
```

Comment:
I would prefer not to use a default capture, `[&]`. Instead I think it could be `[this, threadLoop]`. This looks preferable to me, because it treats `threadLoop` inside the lambda as a `const unsigned` value, instead of a non-const reference. See also Scott Meyers - Effective Modern C++ item "Avoid default capture modes"

Reply:
Yes, I began pondering the best ways to specify the captures with reference vs value. And thinking about the cost of passing lambdas (and other objects) by value vs reference which are done in several places. Do you have and thoughts on how these captures should be pass?

I'll update the PR to explicitly specify the captures.

Reply:
@blowekamp 
> Do you have and thoughts on how these captures should be pass?

I _think_ it's like this:
* If you need to modify the object inside the lambda, do capture by reference `[&obj]`.
* If the object has an expensive copy-constructor, consider doing capture by reference `[&obj]`.
* Otherwise, do capture by value `[obj]`.

Capture "by value" is often more safe than capture "by reference", because the value is treated as `const` value within the lambda (unless it's a `mutable` lambda, of course). When doing  capture "by reference" you must ensure that the reference isn't _dangling_ once the lambda is executed. 

Note that if you capture a (smart) pointer by value, you may still modify the object it points to!

Regarding the cost, I'm afraid you'd just have to measure!

> I'll update the PR to explicitly specify the captures.

Thanks, cool  😃

Reply:
For these particular cases where the lambda is executed immediately, there is not a risk of dangling references.